resultsdbupdater/utils.py

```diff
@@ -236,25 +235,58 @@ def handle_ci_umb(msg):
     if isinstance(system, list):
         system = system[0] if system else {}
 
-    result_data = {
-        'item': item,
-        'type': item_type,
-        'brew_task_id': brew_task_id,
-        'category': msg_body['category'],
-        'component': component,
-        'scratch': scratch,
-        'issuer': msg_body_artifact.get('issuer'),
-        'rebuild': msg_body['run'].get('rebuild'),
-        'log': msg_body['run']['log'],  # required
-        'system_os': system.get('os'),
-        'system_provider': system.get('provider'),
-        'ci_name': msg_body_ci.get('name'),
-        'ci_url': msg_body_ci.get('url'),
-        'ci_environment': msg_body_ci.get('environment'),
-        'ci_team': msg_body_ci.get('team'),
-        'ci_irc': msg_body_ci.get('irc'),
-        'ci_email': msg_body_ci.get('email'),
-    }
+    if item_type == "compose":
+        result_data = {
+            key: value for key, value in (
+                ('ci_name', msg_body['ci']['name']),
+                ('ci_team', msg_body['ci']['team']),
+                ('ci_url', msg_body['ci']['url']),
+                ('ci_irc', msg_body['ci'].get('irc')),
+                ('ci_email', msg_body['ci']['email']),
+
+                ('log', msg_body['run']['log']),
+
+                ('type', item_type),
+                ('productmd.compose.id', msg_body['artifact']['compose_id']),
+
+                ('system_provider', system['provider']),
+                ('system_architecture', system['architecture']),
+                ('system_variant', system.get('variant')),
+
+                ('category', msg_body.get('category')),
+            ) if value is not None
```

Comment:
Optional: I'm wondering if it's better to have consistency and just put null values for keys that aren't present in the message but should be. What do you think?

Reply:
Resultsdb doesn't accept null values for the extra data though, does it? It is a mapping string -> list of string.